crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2184,6 +2184,53 @@ impl World {
                 .get_mut_by_id(component_id)
         }
     }
+
+    /// Iterates over all resources in the world
```

Comment:
I'd like to see some doc tests here: the usage of these two methods will be very hard to piece together from the docs alone, and very easy to break.

Reply:
Added the doctests.